community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -482,10 +477,9 @@ public NodePropertyExistenceConstraint nodePropertyExistenceConstraintCreate( Ke
     public RelationshipPropertyExistenceConstraint relationshipPropertyExistenceConstraintCreate( KernelStatement state,
             int relTypeId, int propertyKeyId ) throws AlreadyConstrainedException, CreateConstraintFailureException
     {
-        try ( Cursor<RelationshipItem> cursor = relationshipCursorGetAll( state ) )
-        {
-            constraintSemantics.validateRelationshipPropertyExistenceConstraint( cursor, relTypeId, propertyKeyId );
-        }
+        Iterator<Cursor<RelationshipItem>> relationships = new EntityLoadingIterator<>( relationshipsGetAll( state ),
+                (id) -> relationshipCursorById( state, id ) );
+        constraintSemantics.validateRelationshipPropertyExistenceConstraint( relationships, relTypeId, propertyKeyId );
```

Comment:
The `AllStoreIdIterator` does not load the record; it only counts through `0` to `highId`.